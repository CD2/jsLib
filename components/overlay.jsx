import React from 'react'
import PropTypes from 'prop-types'
import { styled } from 'utils/theme'

@styled`

  height: 100%;
  left: 0;
  opacity: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 4999;

  &.visible {
    background: $black;
    cursor: pointer;
    opacity: .4;
  }


`
export default class overlay extends React.Component {

  static propTypes = {
    onClick: PropTypes.func.isRequired,
    visible: PropTypes.bool.isRequired,
    className: PropTypes.string,
  }

  static defaultProps = {
    visible: false,
    className: '',
  }


  render() {
    let className = this.props.className
    if (this.props.visible) className += ' visible'

    return (<div className={className} onClick={this.props.onClick}/>)
  }

}
