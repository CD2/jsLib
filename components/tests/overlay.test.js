import t from 'utils/test_helper'

import {Overlay} from '../overlay';

t.describe_component(Overlay, {onClick:() => {}, visible:true, className:'test'}, (c) => {
  describe('proptypes', () => {
    c.accepts_prop('onClick')
    c.accepts_prop('visible')
    c.accepts_prop('className')
  })
  c.has_css('.test .visible')
})
