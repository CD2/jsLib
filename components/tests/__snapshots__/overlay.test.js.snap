// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Overlay /> renders correctly 1`] = `"<div class=\\"test visible\\"></div>"`;
